distributed/client.py

```diff
@@ -1163,22 +1163,20 @@ async def _start(self, timeout=no_default, **kwargs):
         elif self._start_arg is None:
             from distributed.deploy import LocalCluster
 
+            local_cluster = partial(
+                LocalCluster,
+                loop=self.loop,
+                asynchronous=self._asynchronous,
+                **self._startup_kwargs,
+            )
+
             try:
-                self.cluster = await LocalCluster(
-                    loop=self.loop,
-                    asynchronous=self._asynchronous,
-                    **self._startup_kwargs,
-                )
+                self.cluster = await local_cluster()
             except OSError as e:
                 if e.errno != errno.EADDRINUSE:
                     raise
                 # The default port was taken, use a random one
-                self.cluster = await LocalCluster(
-                    scheduler_port=0,
-                    loop=self.loop,
-                    asynchronous=True,
-                    **self._startup_kwargs,
-                )
+                self.cluster = await local_cluster(scheduler_port=0)
```

Comment:
Hrm, one moment.  What happens if `self._asynchronous` is `False`?  Will this try to start up a new LocalCluster in a separate event loop?

Reply:
the LocalCluster doesn't end up running on a new loop in a new thread because we pass the currently running loop as the loop kwarg here:

`LocalCluster(loop=self.loop, asynchronous=False, ...)` calls `SpecCluster(loop=loop, asynchronous=False, ...)`


SpecCluster only starts a new loop in a new thread `if not self.asynchronous`
https://github.com/dask/distributed/blob/5feb17151cdf660a3443abf8596444a9f51dc575/distributed/deploy/spec.py#L258-L265
`self.asynchronous is True` because `self.loop.asyncio_loop is asyncio.get_running_loop()`
https://github.com/dask/distributed/blob/5feb17151cdf660a3443abf8596444a9f51dc575/distributed/utils.py#L303-L334